src/wp-includes/rest-api/endpoints/class-wp-rest-block-directory-controller.php

```diff
@@ -0,0 +1,514 @@
+	 *
+	 * Only the 'Direct' filesystem transport, and SSH/FTP when credentials are stored are supported at present.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @return bool|WP_Error True if filesystem is available, WP_Error otherwise.
+	 */
+	private static function is_filesystem_available() {
+		$filesystem_method = get_filesystem_method();
+
+		if ( 'direct' === $filesystem_method ) {
+			return true;
+		}
+
+		ob_start();
+		$filesystem_credentials_are_stored = request_filesystem_credentials( self_admin_url() );
+		ob_end_clean();
+
+		if ( $filesystem_credentials_are_stored ) {
+			return true;
+		}
+
+		return new WP_Error( 'fs_unavailable', __( 'The filesystem is currently unavailable for installing blocks.' ) );
+	}
+
+	/**
+	 * Parse block metadata for a block, and prepare it for an API repsonse.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param WP_Object $plugin The plugin metadata.
+	 * @param WP_REST_Request $request Request object.
+	 * @return WP_Error|WP_REST_Response Response object on success, or WP_Error object on failure.
+	 */
+	public function prepare_item_for_response( $plugin, $request ) {
+
+		// There might be multiple blocks in a plugin. Only the first block is mapped.
+		$block_data   = reset( $plugin['blocks'] );
+
+		// A data array containing the properties we'll return.
+		$block = array(
+			'name'                  => $block_data['name'],
+			'title'                 => ( $block_data['title'] ? $block_data['title'] : $plugin['name'] ),
+			'description'           => wp_trim_words( $plugin['description'], 30, '...' ),
+			'id'                    => $plugin['slug'],
+			'rating'                => $plugin['rating'] / 20,
+			'rating_count'          => intval( $plugin['num_ratings'] ),
+			'active_installs'       => intval( $plugin['active_installs'] ),
+			'author_block_rating'   => $plugin['author_block_rating'] / 20,
+			'author_block_count'    => intval( $plugin['author_block_count'] ),
+			'author'                => wp_strip_all_tags( $plugin['author'] ),
+			'icon'                  => ( isset( $plugin['icons']['1x'] ) ? $plugin['icons']['1x'] : 'block-default' ),
+			'assets'                => array(),
+			'humanized_updated'     => sprintf(
+				/* translators: %s: Human-readable time difference. */
+				__( '%s ago' ),
+				human_time_diff( strtotime( $plugin['last_updated'] ), current_time( 'timestamp' ) )
+			),
+		);
+
+		foreach ( $plugin['block_assets'] as $asset ) {
+			// TODO: Return from API, not client-set.
+			$block[ 'assets' ][] = 'https://plugins.svn.wordpress.org/' . $plugin['slug'] . $asset;
+		}
+
+		$response = new WP_REST_Response();
+		$response->set_data( $block );
+
+		return $response;
+	}
+
+	/**
+	 * Retrieves the theme's schema, conforming to JSON Schema.
+	 *
+	 * @since 5.5.0
+	 *
+	 * @return array Item schema data.
+	 */
+	public function get_item_schema() {
+		$schema = array(
+			'$schema'    => 'http://json-schema.org/draft-04/schema#',
+			'title'      => 'block',
```

Comment:
```suggestion
			'title'      => 'block-directory-item',
```

Or something similar. We'll have a blocks endpoint at some point which wouldn't have the same schema.